lib/refinement/analyzer.rb

```diff
@@ -57,9 +57,10 @@ def annotate_targets!
     #   `:building` and `:testing`, with the only difference being `BuildActionEntry` are not
     #   filtered out when building for testing, since test action macro expansion could
     #   depend on a build entry being present.
-    # @return [Xcodeproj::XCScheme] a scheme whose unchanged targets have been removed
+    # @param each_target [Boolean] A Proc called each time a target was determined to have changed or not.
+    # @return [Xcodeproj::XCScheme] a scheme whose unchanged targets have been removed.
     def filtered_scheme(scheme_path:, change_level: :full_transitive, filter_when_scheme_has_changed: false, log_changes: false,
-                        filter_scheme_for_build_action:)
+                        filter_scheme_for_build_action:, each_target: nil)
```

Comment:
long term, wonder if this should be some sort of `delegate` pattern instead? rather than adding individual hooks. similar to Molinillo

Reply:
maybe that doesn't make sense here because this API is used directly?

Reply:
yeap our internal tooling directly invokes it.